### ISSUE TITLE
`disallowed_macros` false negatives for specific derive macros

### ISSUE THREAD

**Issue by tgross35** [rust-lang/rust-clippy#12254]
### Summary

`disallowed_macros` seems to have issues flagging specific derive macros such as `async_graphql::SimpleObject`. I'm not sure why, because trying the same below example with `serde::Serialize` does work correctly.

### Lint Name

disallowed_macros

### Reproducer

I tried this code:

```rust
#![deny(clippy::disallowed_macros)]

use async_graphql::SimpleObject;

#[derive(Debug, SimpleObject)]
struct Foo {
    a: i32,
}

fn main() {
    println!("Hello, world!");
}
```

```toml
# clippy.toml
disallowed-macros = [
    { path = "async_graphql::SimpleObject", reason = "no serializing" },
]
```

I expected to see this happen: clippy fails the check

Instead, this happened: no warnings or errors

### Version

```text
rustc 1.77.0-nightly (11f32b73e 2024-01-31)
binary: rustc
commit-hash: 11f32b73e0dc9287e305b5b9980d24aecdc8c17f
commit-date: 2024-01-31
host: aarch64-apple-darwin
release: 1.77.0-nightly
LLVM version: 17.0.6
```

**Comment by Alexendoo** [comment]
[The expansion](https://docs.rs/async-graphql-derive/7.0.1/src/async_graphql_derive/simple_object.rs.html#374) contains `#[allow(clippy::all)]`, so the lint is suppressed

**Comment by tgross35** [comment]
Wow, thanks. It seems like the lint should ignore this sort of thing if the type exists before expansion, no?

No clue if that is even feasible. Opened an issue with the library anyway https://github.com/async-graphql/async-graphql/issues/1471.

**Comment by Centri3** [comment]
That might be an issue if smth like `vec!` is disallowed yet is called in the expanded code
